utils/common/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -87,8 +87,12 @@ fun archive(inputDir: File, zipFile: File, prefix: String = ""): File =
 /**
  * Unpack the [File] to [targetDirectory] using [filter] to select only the entries of interest.
  */
-fun File.unpack(targetDirectory: File, filter: (ArchiveEntry) -> Boolean = { true }) =
-    when (ArchiveType.getType(name)) {
+fun File.unpack(
```

Comment:
Commit message nits:
- No comma in "To support cases, in which"
- "there is valid file name" -> "there is *a* valid file name"
- "to derive the archive type" -> "to derive the archive type *from*"

Maybe better after all: "To support cases where no archive type can be derived from the file name, allow to pass an explicit archive type fallback."

Reply:
Reworded and adapted to the override logic.